File: schemas/helpdesk-customer-support/convex/ticket_followers.ts
// ticket_followers: users subscribed to updates on a specific ticket.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const ticket_followers = defineTable({
  ticketId: v.id("tickets"),
  userId: v.id("users"),
})
  .index("by_ticket_id_and_user_id", ["ticketId", "userId"])
  .index("by_user_id", ["userId"]);

File: schemas/helpdesk-customer-support/spacetimedb/ticket_followers.rs
// ticket_followers: Users subscribed to ticket updates.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = ticket_followers, public)]
pub struct TicketFollower {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub ticket_id: String, // UUID — FK → tickets.id (cascade delete)
    #[index(btree)]
    pub user_id: String, // UUID — FK → users.id (cascade delete)
    // Composite unique: (ticket_id, user_id) — not supported inline, enforce in application
    pub created_at: Timestamp,
}
